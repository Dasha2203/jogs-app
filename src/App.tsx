import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import AppProvider from './context/AppContext';
import HomePage from './pages/HomePage';
import InfoPage from './pages/InfoPage';
import JogsPage from './pages/JogsPage';
import NotFoundPage from './pages/NotFoundPage';

const router = createBrowserRouter([
  {
    path: '/',
    element: <HomePage />,
    errorElement: <NotFoundPage />,
  },
  {
    path: '/jogs',
    element: <JogsPage />,
  },
  {
    path: '/info',
    element: <InfoPage />,
    errorElement: <NotFoundPage />,
  },
]);

function App() {

  return (
    <AppProvider>
      <RouterProvider router={router} />
    </AppProvider>
  );
}

export default App;
