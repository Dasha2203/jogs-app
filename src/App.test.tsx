import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

describe("Test App", () => {
  test("Check text Hello world", async () => {
    render(<App />);
    const text = screen.queryByText(/Hello world/i);
    expect(text).toBeNull();
  });

  test('Check "Let me in"', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
    const textBtn = 'Let me in';
    const button = screen.getByRole('button');

    expect(button).toHaveTextContent(textBtn);
    expect(button).toMatchSnapshot();
  })
});