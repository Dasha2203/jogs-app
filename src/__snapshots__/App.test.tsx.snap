// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Check "Let me in" 1`] = `
<button
  class="button button_lg"
  type="button"
>
  Let me in
</button>
`;
